plutus-core/plutus-ir/src/PlutusIR/Transform/Inline.hs

```diff
@@ -447,10 +476,13 @@ For non-strict bindings, the effects already happened at the use site, so it's f
 unconditionally.
 -}
 
+-- | Check against the inlining heuristics for types and either inline it, returning Nothing, or
+-- just return the type without inlining.
+-- We only inline if (1) the type is used at most once OR (2) its a `trivialType`.
 maybeAddTySubst
     :: forall tyname name uni fun a . InliningConstraints tyname name uni fun
-    => tyname
-    -> Type tyname uni a
+    => tyname -- ^ The type name.
+    -> Type tyname uni a -- ^ Its type.
```

Comment:
This is confusing, I wouldn't say that the type "belongs" to the type variable. Both the name and the type belong to a _binding_. Or you could say "a type variable and the value of the variable"